charts/kong/README.md

```diff
@@ -288,6 +293,124 @@ scenario and several settings that are useful when using multiple controllers:
 Standalone controllers require a database-backed Kong instance, as DB-less mode
 requires that a single controller generate a complete Kong configuration.
 
+### Hybrid mode
+
+Kong supports [hybrid mode
+deployments](https://docs.konghq.com/2.0.x/hybrid-mode/) as of Kong 2.0.0 and
+Kong Enterprise 2.1.0. These deployments split Kong nodes into control plane
+(CP) nodes, which provide the admin API and interact with the database, and
+data plane (DP) nodes, which provide the proxy and receive configuration from
+control plane nodes.
+
+The chart does not support deploying control and data plane nodes in a single
+release. Users should [create separate releases for each node
+type](#separate-admin-and-proxy-nodes), i.e. use separate control and data
+plane values.yamls that are installed separately.
+
+* If `.Values.env.role="control_plane"`, the release will create a Service
+  named `RELEASE-NAME-kong-cluster`.
+* Cluster certificates are not yet generated automatically. You must create a
+  certificate/key pair from a local instance, create a Secret containing them,
+  add the Secret to `.Values.secretVolumes`, and add
+  `cluster_cert`/`cluster_cert_key` values indicating its location under
+  `/etc/secrets` to `env`.
+* Note that data plane nodes do not default to using the default port (8005)
+  for `cluster_control_plane`. You should use `RELEASE-NAME-kong-cluster:8005`
+  as the `cluster_control_plane` value if using the default port.
+
+#### Certificates
+
+Hybrid mode uses TLS to secure the CP/DP node communication channel, and
+requires certificates for it. You can generate these either using `kong hybrid
+gen_cert` on a local Kong installation or using OpenSSL:
```

Comment:
Suggested reword to avoid ambiguous pronouns and to specify the course of action:

> Hybrid mode secures the CP/DP node communication channel using TLS with client and server certificates as means of authentication. In order to deploy Kong in hybrid mode, you need to:
> 1. generate appropriate certificates,
> 1. wrap the newly generated certificates in Kubernetes secrets and apply them to the cluster, and
> 1. tell the Helm chart where the certificates are by specifying secret names in chart values.

Please verify the suggestion for factual correctness.

Reply:
Agree that the "it" in the first sentence is ambiguous, but I don't think we need to try and clarify further--whether the reference is back to "Hybrid mode", "communication channel", or "TLS" isn't crucial to understand (arguably, any interpretation is correct); the main point is that you need certs.

The rest I try to show through the exact commands and configuration used to reach the end state, i.e. if you use the `openssl` and `kubectl create secret` examples as shown, the values.yaml examples later will load the resulting certificate Secret.

Reply:
My main concern here is that this section leaves the user with an understanding that they need to generate and upload certificates, but (if the user doesn't have enough willpower to read the following section as well) there will be no mention of a need to point the helm release to the uploaded secret. This is not obvious for a reader IMO.

I suggest at the very least making a note at the end of this section (_Certificates_) to read that the user should also set the secret name as described in the following section (_Control plane node configuration_)

What would fix the problem is a simple reference to the following paragraph. If you :

> Once you apply the certificate secret to the cluster, please specify the secret name in `values.yaml` (see the section below)"

Reply:
That level of detail and the linear step order would be a good candidate for the guide section mentioned in #164. For this document I prefer to present the requirements briefly through examples that work together and build on it as we receive questions from users, rather than try to account for every possible hypothetical question in advance.

Reply:
The problem my comment originated from is that I read this section as inaccurate. It is not sufficient to **generate and apply** certs in order to use the hybrid mode. It is required and sufficient to **pass existing cert secret names to the chart**. Right? Currently this section says the former and I believe it should say the latter.

Current wording feels misleading to me (it may leave the user with an assumption that the chart can „find” the cert secret somehow, for example if the name follows some convention.